### ISSUE TITLE
[Linux] IDirect3DQuery9::GetData(): Occlusion query flush took 8.544ms!

### ISSUE THREAD

**Issue by TheBatman** [ValveSoftware/Source-1-Games#1936]
IDirect3DQuery9::GetData(): Occlusion query flush took 0.544ms! My console keeps getting spammed with that and i'm suffering FPS drops everytime this happens.

here the system info:

Processor Information:
    Vendor:  AuthenticAMD
    CPU Family:  0x16
    CPU Model:  0x30
    CPU Stepping:  0x1
    CPU Type:  0x0
    Speed:  2000 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    Ubuntu 14.04.3 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.19.0-33-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11701000
    X Window Manager:  Compiz
    Steam Runtime Version:  steam-runtime-release_2015-06-12
    
Video Card:
    Driver:  ATI Technologies Inc. AMD Radeon(TM) R5 Graphics        

    Driver Version:  4.5.13399 Compatibility Profile Context 15.20.1013
    OpenGL Version: 4.5
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x1002
    DeviceID:  0x9851
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1366 x 768
    Desktop Resolution: 1366 x 768
    Primary Display Size: 13.54" x 7.60"  (15.51" diag)
                                            34.4cm x 19.3cm  (39.4cm diag)
    Primary VRAM Not Detected
    
Sound card:
    Audio device: ATI R6xx HDMI
    
Memory:
    RAM:  6848 Mb
    
Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  461830 Mb
    Largest Free Hard Disk Block:  367537 Mb
    
Installed software:
    
Recent Failure Reports:

**Comment by TheBatman** [comment]
![36324948-a6a0-11e5-9091-2364fc854409](https://cloud.githubusercontent.com/assets/13712357/11986843/1da39646-a99a-11e5-8cbe-3df44337c173.jpg)

Help.

**Comment by Nephyrin** [comment]
This sounds like a symptom of a driver or compositor bug of some kind. Something is causing the GPU to stall for long periods.

I would try disabling compiz and any other GPU using apps, and ensuring your drivers are up to date.

**Comment by TheBatman** [comment]
How can I disable compiz? I was using the latest drivers, I find this
problem mostly in source games but 3rd party games (most) ran smooth.

On Feb 9, 2016 1:49 PM, "John Schoenick" <notifications@github.com> wrote:
>
> This sounds like a symptom of a driver or compositor bug of some kind.
Something is causing the GPU to stall for long periods.
>
> I would try disabling compiz and any other GPU using apps, and ensuring
your drivers are up to date.

**Comment by MRebhan** [comment]
I have the same problem, but for me there are no fps drops, I'm running at 140 fps constant (TF2)

Arch Linux x64, R9 380 using catalyst/fglrx driver.
Maybe it is a AMD problem?
@TheBatman are you using open-source driver or catalyst
I'm guessing catalyst judging by the graphics card info

**Comment by TheBatman** [comment]
@MRebhan maybe if i try Arch Linux my problem will get solved since my only concern are the drops.

how can i install Arch?

**Comment by TheBatman** [comment]
@MRebhan btw i am using Open source ones right now, but the problem happens if i have fglrx drivers installed

**Comment by jrdoane** [comment]
I was able to eliminate the stuttering and message in the console by disabling multi-core rendering under the advanced options in the video tab for settings. Is it possible that the way the source engine does this is incompatible with both AMDGPU(-Pro) and open source Radeon drivers causing parts of the 3D pipeline to stall? I'm using Ubuntu 16.04 with Unity so, Compiz is active as well. I have a Radeon R9 390 with AMDGPU-Pro.

**Comment by TheBatman** [comment]
@jrdoane i don't have Linux installed right now, how does the new open source drivers work?

**Comment by jrdoane** [comment]
@TheBatman It's very similar to the radeon driver from what I can tell in Ubuntu 16.04. There are several situations where AMDGPU-Pro seems more stable and faster than the radeon open source driver. I've found that in general it has resulted in a better experience but, the drivers are still pretty immature. It's **almost** good enough for every-day use but, there are a handful of cases where I would just like to have Windows-like performance again.

Big warning on AMDGPU (open source by itself,) is that it's only supported on GCN 1.2 cards. There is only experimental support for GCN 1.1 which requires recompiling the kernel with a flag set. (This can be avoided by using AMDGPU-Pro which includes AMD's proprietary stuff.) All in all, it's a work in progress but, it's most definitely getting better. If you're a gamer, I wouldn't jump off the Windows ship quite yet though.

**Comment by TheBatman** [comment]
@jrdoane i will have to stay in windows for more time then, thanks for the update about AMD hope we can see the same or better perfomance than in windows with Linux OS :+1:

**Comment by doug65536** [comment]
Same thing happens on Ryzen 7 using GTX 1060 with nvidia-375 driver on Ubuntu 16.04 and 17.04.

**Comment by menne386** [comment]
Also happens on core i5 with GTX 1060 and latest linux drivers.... would occlusion_test_async help here? will test out when home

**Comment by MRebhan** [comment]
yeah after getting a GTX 1070 it also happens there, but still no fps drops for me.
